client/my-sites/importer/site-importer/site-importer-input-pane.jsx

```diff
@@ -0,0 +1,243 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import Dispatcher from 'dispatcher';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import React from 'react';
+import { noop, every, has, defer, get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wpLib from 'lib/wp';
+const wpcom = wpLib.undocumented();
+
+import { toApi, fromApi } from 'lib/importer/common';
+
+import { startMappingAuthors, startImporting, mapAuthor, finishUpload } from 'lib/importer/actions';
+import user from 'lib/user';
+
+import { appStates } from 'state/imports/constants';
+import Button from 'components/forms/form-button';
+import ErrorPane from '../error-pane';
+import TextInput from 'components/forms/form-text-input';
+
+import SiteImporterSitePreview from './site-importer-site-preview';
+import { connectDispatcher } from '../dispatcher-converter';
+
+class SiteImporterInputPane extends React.Component {
+	static displayName = 'SiteImporterSitePreview';
+
+	static propTypes = {
+		description: PropTypes.oneOfType( [ PropTypes.node, PropTypes.string ] ),
+		importerStatus: PropTypes.shape( {
+			importerState: PropTypes.string.isRequired,
+			percentComplete: PropTypes.number,
+		} ),
+		onStartImport: PropTypes.func,
+		disabled: PropTypes.bool,
+	};
+
+	static defaultProps = { description: null, onStartImport: noop };
+
+	state = {
+		// TODO this is bad, make it better. Both appStates and state should be unified in one.
+		importStage: 'idle',
+		error: false,
+		errorMessage: '',
+		siteURLInput: '',
+	};
+
+	// TODO This can be improved if we move to Redux.
```

Comment:
This fn looks awful with those defers but agreed that we can handle it better once switching to redux.